src/network/networkd-link.c

```diff
@@ -2619,16 +2619,32 @@ static int link_set_ipv6_mtu(Link *link) {
 }
 
 static int link_drop_foreign_config(Link *link) {
-        Address *address;
+        Address *address, *net_address;
         Route *route;
         Iterator i;
         int r;
+        bool skip_address;
```

Comment:
please move these two additions to the innermost scope this is needed, i.e. into the if-block you are adding. we generally follow the rule of declaring variables with minimal scope (with the exception of `int r`, which is so frequently used for error handling that it gets a special status)